### PR TITLE
feat: addtocart

### PR DESCRIPTION
**Pull request by naveen-pandey**

### CODE REVIEW

File: src/pages/Products/ProductDetail.js [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
+      <img src={currentProduct.image} alt={currentProduct.title} />
+      <p>{currentProduct.description}</p>
+      <p>${currentProduct.price}</p>
+      <button onClick={handleAddToCart} className="add-to-cart-btn">Add to Cart</button>
     </div>
   );
 };
 
-export default ProductDetail;
+export default ProductDetail;
```
Comment (nishantt12) [inline src/pages/Products/ProductDetail.js:45]: Add new line at the end of the file.

**Comment by nishantt12** [comment]
Please follow best practices for the PR:

1. Changes are going
2. Any JIRA ticket
3. How did you test the changes?

File: src/pages/Products/ProductDetail.js [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
 
   const handleAddToCart = () => {
-    console.log('Product added to cart:', product);
+    dispatch(addToCart(currentProduct));
+    navigate('/dashboard/cart');
   };
 
-  if (!product) {
-    return <div className="loading">Loading...</div>;
+  if (status === 'loading') {
```
Comment (nishantt12) [inline src/pages/Products/ProductDetail.js:21]: We should not use hardcoded values.

File: src/pages/Products/ProductDetail.js (hunk 1 above)
Comment (naveen-pandey) [reply src/pages/Products/ProductDetail.js:45]: Ok

**Pull request closed without merging**